Modules, Exports, and Imports.md

```diff
@@ -0,0 +1,301 @@
+blocks.
+
+## Packages and Applications
+
+Elm programs fall into two categories. **Packages** (or **libraries**) are meant to be reused and are typically
+published to the [package registry](http://package.elm-lang.org/). **Applications** are programs that can actually be
+run and produce output. Applications may depend on packages, which may depend on other packages in turn; applications
+are the root of this tree.
+
+The distinction between package and application occurs in `elm-package.json`: A package will export at least one module
+in the `"exposed-modules"` field. Packages are compiled with `elm make` and no additional arguments. This compiles the
+modules listed, and subjects them to additional compile-time checks ensuring every exported value and type are
+documented properly. The compiler does not generate an HTML or JS output file.
+
+If you pass a file to `elm make`, then you are writing an application. That file (and therefore module) is known as
+"Main". It's not necessary to call this file `Main.elm` or even declare a module definition in it at all -- but you
+should for large applications.
+
+However defined, only Main can define ports (a topic otherwise beyond the scope of this guide). The Main module must
+define a `main` value, whose type is restricted to a handful of things that the compiler can display. Other modules
+should avoid defining `main`.
+
+Another difference between packages and applications is how they tolerate change. Elm's package manager enforces
+[semantic versioning](http://semver.org/), discouraging breaking changes. Additionally anything that is exported must be
+documented. Packages therefore want to hide their type definitions and helper functions, in order to present a stable,
+understandable, and helpful interface. Applications are more tolerant of change; for example adding a field to a record
+type alias or a tag to a union type. Only the application itself relies on these definitions, and the compiler will find
+any discrepancies that changes introduce. Therefore, packages must tightly control what is exported, but applications
+can afford to be more permissive.
+
+## Modules and Files
+
+Every file contains exactly one module. Filenames should be capitalized and match the name of the module they contain.
+For example, `Foo.elm` should declare the module `Foo`. The compiler will not allow module names and file paths to
+disagree.
+
+If you only have one file, it's fine to keep it in the top level folder. Otherwise, you should create a `src` folder and
+keep all source files in there. You'll need edit `elm-package.json` to read `"source-directories": [ "src" ],` (don't
+forget the comma; `elm package` doesn't give a helpful error on invalid JSON).
+
+Module names can contain dots, for example `Json.Encode`, which is in file `src/Json/Encode.elm`. Folders may be nested
+as deep as you like, but don't overdo it. Sometimes a package will have a primary package `Foo` and a secondary package
```

Comment:
`Sometimes a package will have a primary package ...`
do you define Json.Encode as a package? To me is just a module inside a package